### DISCUSSION TITLE
blog/2019-10-31-time-varying-covariates-in-longitudinal-multilevel-models-contain-state-and-trait-level-information-this-includes-binary-variables-too/

### DISCUSSION THREAD

**Discussion by giscus** [ASKurz/blogdown5#23] [Comments]
# blog/2019-10-31-time-varying-covariates-in-longitudinal-multilevel-models-contain-state-and-trait-level-information-this-includes-binary-variables-too/

When you have a time-varying covariate you'd like to add to a multilevel growth model, it's important to break that variable into two. One part of the variable will account for within-person variation. The other part will account for between person variation. Keep reading to learn how you might do so when your time-varying covariate is binary.

https://solomonkurz.netlify.app/blog/2019-10-31-time-varying-covariates-in-longitudinal-multilevel-models-contain-state-and-trait-level-information-this-includes-binary-variables-too/

**Comment by DrJerryTAO** [comment]
Several of the display equations cannot render correctly. Could you fix them?

You recommended to "Leave the level-1 version alone." I wonder whether it makes sense to find the difference between the individual mean and the time-specific values. In your case, it is to replace `unemp` with `unemp - unemp_id_mu`. Does it make sense to add interaction between the individual mean and the time-specific values, such as `unemp : unemp_id_mu` or `unemp : I(unemp - unemp_id_mu)`?

Your data glimpse shows that `months` is also a time-varying predictor, yet you did not apply the same approach used for unemployment to `months`. Was it an error? `fit2` has introduced one interaction term `months : unemp_id_mu`, but with `months` recognized as a time-varying predictor just like `unemp`, how should the necessary interaction terms change? I think one and only one interaction term might suffice, which is `months_id_mu : unemp_id_mu`, if main effects of both `months_id_mu` and `unemp_id_mu` are retained. The interaction `months : unemp` seems possible, but does the modelling technique support estimating this interaction? 

With the individual means of months and unemployment probabilities added, how should one interpret the effect of unemployment on depression? Should one use `gamma_{20}` directly? This coefficient seems to quantify how depression changes if an employed worker becomes unemployed with their unemployment probability controlled for. But it is plausible that this change depends on the worker's unemployment probability: historically often employed worker may have a huge depression increase if unemployed, whereas a historically often unemployed worker may not be surprised by another time of unemployment. Thus, adding interaction between the individual mean and the time-specific unemployment measure, such as `unemp : unemp_id_mu` or `unemp : I(unemp - unemp_id_mu)`, should be beneficial. But does the modelling technique support estimating this interaction?

How much potential do longitudinal non-experimental data have for identifying causal effects? The natural effect interpretation is "for any person who were initially unemployed but got employed at any future time t, the depression level decreased by gamma_{20} on average," which may not necessarily carry any causal inference. I understand that no employment was randomly assigned to initially unemployed workers. Could we possibly use the variation in the employment status that changed at different time within the same worker to claim that the above effect can be interpreted causally? The reason is repeated measurements (3 times for each worker) removed effects of individual-specific traits, such as genome traits, demographic characteristics, and family cultures, on depression. 

Your models `fit1` and `fit2` appear to have equivalence in the frequentist multi-level modeling as the random-effect estimator and the fixed-effect estimator. The random-effect estimator assumes zetas to be unrelated to all the predictors, whereas the fixed-effect estimator relaxes this assumption by adding dummy individual identifiers as additional predictors. Adding individual means in a random-intercept and/or random-coefficient model can test whether the random-effect or the fixed-effect estimator should be used. If the individual means are collectively significant, then the fixed-effect estimator should be used, otherwise the more concise random-effect estimator suffices. Is there a random-effect and the fixed-effect assumption in the Bayesian framework? Does the goodness of fit of `fit1` and `fit2` suggest anything regarding whether zetas are related to the predictors?

**Reply by ASKurz** [reply]
Ug. Thanks for the heads up on the un-rendered equations!

Regarding your thoughts on `unemp - unemp_id_mu`, I haven't had a reason to use an approach like that, so you'd have to experiment and see. 

Yes, `months` is a time varying predictor. To my mind, it's a special kind of time-varying predictor in that it's the measure of time in the model, and I don't believe I have ever seen a methodologist propose decomposing the time variable before. If you have, please share the reference.

I haven't explored the literature on non-experimental longitudinal data for causal effects, so I really couldn't comment.

I'm not a fan of the fixed-effects approach to longitudinal data analysis.
